src/platforms/esp32/components/avm_builtins/Kconfig

```diff
@@ -38,9 +38,12 @@ config AVM_ENABLE_RTC_SLOW_NIFS
 
 config AVM_RTC_SLOW_MAX_SIZE
     int "RTC SLOW maximum binary size"
-    # By default, use the maximum size for esp-idf < 4.4
-    # In newer version, available size is 8170
-    default 4086
+    # Maximum size for esp-idf < 4.4 is 4086
+    # Newer versions of the SDK allow up to 8170
+    # ESP32-H2 has no RTC SLOW, only RTC FAST and both are mapped at the same
+    # space, and based on other usage of RTC FAST we can use up to 3.5kb
+    # Default should compile on every target.
+    default 3096
```

Comment:
Another option to consider would be to leave the default (or now we can actually raise it to 8096 since 4.4 is the lowest numbered release we build on), and use a sdkcinfig.defaults.esp32h2 that uses the smaller RTC memory size, allowing other targets to still use the larger size.

Reply:
It's better indeed.

Reply:
The maximum size with SDK 5.1 seems to be 8140 (and not 8170) and it may decrease depending on idf versions. So I suggest bumping this to a round 4096, so default doesn't decrease with this PR (and using a separate default file for H2 as you suggested).